client/app/bundles/course/survey/components/ResponseAnswer.jsx

```diff
@@ -0,0 +1,166 @@
+    maxHeight: 150,
+    maxWidth: 400,
+  },
+  optionLabel: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  option: {
+    marginTop: 10,
+  },
+  errorText: {
+    color: red500,
+  },
+};
+
+class ResponseAnswer extends React.Component {
+  static renderTextResponseField(question, member) {
+    return (
+      <Field
+        name={`${member}[text_response]`}
+        component={TextField}
+        style={styles.textResponse}
+        multiLine
+      />
+    );
+  }
+
+  static renderOptionLabel(option) {
+    return (
+      <div style={styles.optionLabel}>
+        { option.image_url ?
+          <Thumbnail src={option.image_url} style={styles.image} /> :
+          []
+        }
+        { option.option ? option.option : '' }
+      </div>
+    );
+  }
+
+  static renderMultipleChoiceOptions({ question, ...props }) {
+    const { meta: { dirty, error } } = props;
+    return (
+      <div>
+        { dirty && error ? <p style={styles.errorText}>{error}</p> : null }
+        <RadioButtonGroup {...props}>
+          {question.options.map(option => (
+            <RadioButton
+              key={option.id}
+              value={option.id.toString()}
+              label={ResponseAnswer.renderOptionLabel(option)}
+              labelStyle={styles.label}
+              style={styles.option}
+            />
+          ))}
+        </RadioButtonGroup>
+      </div>
+    );
+  }
+
+  static renderMultipleChoiceField(question, member) {
+    return (
+      <Field
+        name={`${member}[selected_option]`}
+        component={ResponseAnswer.renderMultipleChoiceOptions}
+        {...{ question }}
+      />
+    );
+  }
+
+  static renderMultipleResponseOptions(props) {
+    const { fields, options, meta: { dirty, error } } = props;
+    return (
+      <div>
+        { dirty && error ? <p style={styles.errorText}>{error}</p> : null }
+        {
+          fields.map((member, index) => {
+            const answerOption = fields.get(index);
+            const option = options.find(opt =>
+              opt.id.toString() === answerOption.question_option_id.toString()
+            );
+            return (
+              <Field
+                key={option.id}
+                name={`${member}[selected]`}
+                component={Checkbox}
+                label={ResponseAnswer.renderOptionLabel(option)}
+                labelStyle={styles.label}
+                style={styles.option}
+              />
+            );
+          })
+        }
+      </div>
+    );
+  }
+
+  static renderMultipleResponseField(question, member) {
+    return (
+      <FieldArray
+        name={`${member}[options]`}
+        component={ResponseAnswer.renderMultipleResponseOptions}
+        options={question.options}
+      />
+    );
+  }
+
+  render() {
+    const { TEXT, MULTIPLE_CHOICE, MULTIPLE_RESPONSE } = questionTypes;
+    const { question, member, index } = this.props;
+
+    if (!question || !question.question_type) { return <div />; }
```

Comment:
Should check whether the type is in the array of types. otherwise when `question.question_type` is `0`, this one will return and nothing get rendered.